### DISCUSSION TITLE
Permission Denied: publikey

### DISCUSSION THREAD

**Discussion by ghost** [community/community#141653] [Programming Help]
### Body

I just got a new PC and after installing git on it, I attempted to clone from one of my GitHub repos using an SSH Key, I encountered this error message:
```
PS C:\Users\jonat> git clone git@github.com:Skrillx13/Skrillx13-Theme.git
Cloning into 'Skrillx13-Theme'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

This error happens with other GitHub Repos as well.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by RanitDERIA** [comment]
If you're encountering the "_**Permission denied (publickey)**_" error when trying to clone a GitHub repository via SSH, it usually indicates that GitHub is unable to authenticate your SSH key. Here are a few steps you can take to resolve this issue:

## 1. Verify SSH Key Setup
Make sure your SSH key is correctly set up on your new PC. You can check if an SSH key already exists by running:

```
ls ~/.ssh/id_rsa.pub
```

If there is no SSH key, you can generate one with:

```
ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
```

Follow the prompts to save the key (usually at `~/.ssh/id_rsa`).

## 2. Add Your SSH Key to the SSH Agent
Make sure the SSH agent is running, and add your SSH key to it:

```
eval "$(ssh-agent -s)"
ssh-add ~/.ssh/id_rsa
```
## 3. Check GitHub Access Permissions
Ensure your GitHub account has the necessary permissions to access the repository. If it's a private repository, make sure you're added as a collaborator.

# Summary
These steps should help you resolve the "Permission denied (publickey)" error. If you continue to encounter issues, double-check the SSH key added to your GitHub account and confirm that it matches the one on your new PC. For further details, refer to [GitHub's SSH troubleshooting guide](https://docs.github.com/en/authentication/troubleshooting-ssh).

_Feel free to ask if you have more questions or need further assistance!_
